libevm/lib/evm_test.go

```diff
@@ -47,26 +48,130 @@ func TestEvmTrace(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	// do a coarse correctness check that does not immediately break on different versions of the solidity compiler
-	if minimum, actual := 130, len(result.TraceLogs); minimum > actual {
-		t.Fatalf("unexpected number of trace logs: expected at least %d, actual %d", minimum, actual)
+}
+
+func TestEvmCallTracer(t *testing.T) {
+	var (
+		instance     = New()
+		err          error
+		initialValue = common.Big0
+		sender       = common.HexToAddress("0xbafe3b6f2a19658df3cb5efca158c93272ff5c0b")
+	)
+	dbHandle := instance.OpenMemoryDB()
+	err, stateDbHandle := instance.StateOpen(StateParams{
+		DatabaseParams: DatabaseParams{
+			DatabaseHandle: dbHandle,
+		},
+		Root: common.Hash{},
+	})
+	if err != nil {
+		t.Fatal(err)
 	}
-	// cherry-pick the one SSTORE instruction that should be in there
-	sstoreInstructions := 0
-	for _, trace := range result.TraceLogs {
-		if trace.Op != "SSTORE" {
-			continue
-		}
-		sstoreInstructions += 1
-		if expected, actual := "SSTORE", trace.Op; expected != actual {
-			t.Fatalf("unexpected op code: expected %s, actual %s", expected, actual)
-		}
-		if expected, actual := 1, len(*trace.Storage); expected != actual {
-			t.Fatalf("unexpected number of accessed storage keys: expected %d, actual %d", expected, actual)
-		}
+	err, _ = instance.EvmApply(EvmParams{
+		HandleParams: HandleParams{
+			Handle: stateDbHandle,
+		},
+		From:  sender,
+		To:    nil,
+		Input: test.StorageContractDeploy(initialValue),
+		Context: EvmContext{
+			Coinbase: common.Address{},
+			BaseFee:  (*hexutil.Big)(new(big.Int)),
+		},
+		TraceOptions: &TraceOptions{
+			EnableMemory:     true,
+			DisableStack:     false,
+			DisableStorage:   false,
+			EnableReturnData: true,
+			Tracer:           "callTracer",
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
```

Comment:
can we do a sanity check of the result structure/data?
Same in other tests

Reply:
Sanity checks added to the Go unit tests, now we check the correct structure of the tracer result and check its values.
I also restore some checks done on the struct logger response present before.
We must take into account that they could fail due to future changes of the native loggers structure, so unrelated to our `libevm` implementation.